docker/scripts/functions.sh

```diff
@@ -403,6 +403,78 @@ function validate_bios_config() {
 	fi
 }
 
+# usage: bios_inventory $id $plan $facility
+function bios_inventory() {
+	local id=$1
+	local plan=$2
+	local facility=$3
+	local hollow_token
+	local hollow_response
+
+	# When running the inventorybios command outside of the packet-hardware
+	# container, UTIL_RACADM7 and UTIL_SUM must be set to the locations of the
+	# racadm and sum binaries, respectively
+	local bios_json_fn="/tmp/bios.json"
+	local hollow_json_fn="/tmp/hollow.json"
+	local hollow_namespace="net.equinixplatform.bios"
+	if UTIL_RACADM7=/opt/dell/srvadmin/bin/idracadm7 UTIL_SUM=/opt/supermicro/sum/sum packet-hardware inventorybios --verbose -u localhost --dry --cache-file "${bios_json_fn}"; then
+		# Generate JSON with additional fields required by Hollow
+		if ! jq --null-input \
+			--arg ns "${hollow_namespace}" \
+			--arg id "${id}" \
+			--slurpfile data "${bios_json_fn}" \
+			'{Namespace: $ns, hardware_uuid: $id, data: $data[]}' >"${hollow_json_fn}"; then
+			echo "Warning: error while generating hollow json"
+			return 0
+		fi
+		echo "BIOS feature inventory JSON is:"
+		cat "${hollow_json_fn}"
+
+		# Disable tracing since we're handling secrets
+		set +x
+
+		# Write data to Hollow
+		if [[ -z ${HOLLOW_CLIENT_ID:-} ]] || [[ -z ${HOLLOW_CLIENT_REQUEST_SECRET:-} ]]; then
+			echo "Warning: env credentials for Hollow not found, not writing to Hollow"
+			set -x
+			return 0
+		fi
+
+		local hollow_auth_url="https://auth.equinixmetal.com/oauth/token"
+		local hollow_audience="https://hollow.platformequinix.net"
+		# Use eclypsium-proxy to reach the auth server from the deprov network
+		echo "Requesting a hollow token from ${hollow_auth_url}"
+		if ! hollow_token=$(HTTPS_PROXY="http://eclypsium-proxy-${facility}.packet.net:8888/" curl --request POST \
```

Comment:
If you change to
```suggestion
		echo "Requesting a hollow token from ${hollow_auth_url}"
		local hollow_token
		if ! hollow_token=$(HTTPS_PROXY="http://eclypsium-proxy-${facility}.packet.net:8888/" curl --request POST \
```
you can bring it down to here, ditto with hollow_url

Reply:
Done.